h2/connection.py

```diff
@@ -684,9 +695,16 @@ def get_next_available_stream_id(self):
         # No streams have been opened yet, so return the lowest allowed stream
         # ID.
         if not self.highest_outbound_stream_id:
-            return 1 if self.config.client_side else 2
+            next_stream_id = 1 if self.config.client_side else 2
```

Comment:
(Optional) As this code unfolds, I wonder if there’s an opportunity to reduce code repetition here. Having an early return in this `if` block made sense when that was the only thing; now we have duplicated code.

I think something like this would be better:

```python
if self.highest_outbound_stream_id:
    ...
else:
    ...

self.log('Next available stream ID')

if next_stream_id > self.HIGHEST_ALLOWED_STREAM_ID:
    ...

return next_stream_id
```